pkg/karmadactl/exec.go

```diff
@@ -93,292 +84,46 @@ func execExample(parentCommand string) string {
 	return example
 }
 
-// RemoteExecutor defines the interface accepted by the Exec command - provided for test stubbing
-type RemoteExecutor interface {
-	Execute(method string, url *url.URL, config *restclient.Config, stdin io.Reader, stdout, stderr io.Writer, tty bool, terminalSizeQueue remotecommand.TerminalSizeQueue) error
-}
-
-// DefaultRemoteExecutor is the standard implementation of remote command execution
-type DefaultRemoteExecutor struct{}
-
-// Execute implement RemoteExecutor interface
-func (*DefaultRemoteExecutor) Execute(method string, url *url.URL, config *restclient.Config, stdin io.Reader, stdout, stderr io.Writer, tty bool, terminalSizeQueue remotecommand.TerminalSizeQueue) error {
-	exec, err := remotecommand.NewSPDYExecutor(config, method, url)
-	if err != nil {
-		return err
-	}
-	return exec.Stream(remotecommand.StreamOptions{
-		Stdin:             stdin,
-		Stdout:            stdout,
-		Stderr:            stderr,
-		Tty:               tty,
-		TerminalSizeQueue: terminalSizeQueue,
-	})
-}
-
-type streamOptions struct {
-	Cluster       string
-	Namespace     string
-	PodName       string
-	ContainerName string
-	Stdin         bool
-	TTY           bool
-	// minimize unnecessary output
-	Quiet bool
-	// InterruptParent, if set, is used to handle interrupts while attached
-	InterruptParent *interrupt.Handler
-
-	genericclioptions.IOStreams
-
-	// for testing
-	overrideStreams func() (io.ReadCloser, io.Writer, io.Writer)
-	isTerminalIn    func(t term.TTY) bool
-}
-
 // ExecOptions declare the arguments accepted by the Exec command
 type ExecOptions struct {
 	// global flags
 	options.GlobalCommandOptions
-
-	streamOptions
-	resource.FilenameOptions
-
-	ResourceName     string
-	Command          []string
-	EnforceNamespace bool
-
-	Builder          func() *resource.Builder
-	ExecutablePodFn  polymorphichelpers.AttachablePodForObjectFunc
-	restClientGetter genericclioptions.RESTClientGetter
-
-	Pod           *corev1.Pod
-	Executor      RemoteExecutor
-	PodClient     coreclient.PodsGetter
-	GetPodTimeout time.Duration
-	Config        *restclient.Config
+	// flags specific to exec
+	KubectlExecOptions *kubectlexec.ExecOptions
+	Namespace          string
```

Comment:
It seems not need this `Namespace` that has include in `StreamOptions`

https://github.com/carlory/karmada/blob/d47f16d6e6edb15ece3c920717864710b25318bd/vendor/k8s.io/kubectl/pkg/cmd/exec/exec.go#L135

Reply:
The namespace field of StreamOptions is obtained from the factory. By default, the namespace value of the context in the configuration file is used. We should allow the user to specify the namespace.

https://github.com/karmada-io/karmada/pull/2092/files#diff-7344b04f1845590962a793044b542d992c24a2e8af35e98bbf614430bb600ddeR193

```shell
(⎈ |karmada:demo)➜  karmada git:(master) go run cmd/karmadactl/karmadactl.go exec -it master-etcd-nodedsp01.ocp.ats.io -C ocp -n kube-system  -- ls
Error: unknown shorthand flag: 'n' in -n
exit status 1

(⎈ |karmada:demo)➜  karmada git:(karmadactl-exec) go run cmd/karmadactl/karmadactl.go exec -it master-etcd-nodedsp01.ocp.ats.io -C ocp -n kube-system  -- ls
member
```

Reply:
I see. Nice finding!